### PR TITLE
Implements JpaRepository, Luquibase, soft delete concept and add missing endpoints for BookService

### PR DESCRIPTION
**Pull request by Extase67**
Replace the existing BookRepository with the one that will use JpaRepository interface
Add Liquibase support
Change spring.jpa.hibernate.ddl-auto property value to validate 
Implement soft delete concept for the Book entity
Add missing endpoints and methods for BookService

### CODE REVIEW

File: src/main/java/com/example/demo/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +34,14 @@ public BookDto findById(@PathVariable Long id) {
     public BookDto save(@RequestBody CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
+
+    @DeleteMapping
+    public void deleteById(@PathVariable Long id) {
```
Comment (JJJazl) [inline src/main/java/com/example/demo/controller/BookController.java:43]: Please, add a corresponding status code for DELETE method with void return type

File: src/main/java/com/example/demo/controller/BookController.java [hunk 2]
```diff
@@ -32,4 +34,14 @@ public BookDto findById(@PathVariable Long id) {
     public BookDto save(@RequestBody CreateBookRequestDto bookDto) {
```
Comment (JJJazl) [inline src/main/java/com/example/demo/controller/BookController.java:37]: Same for POST

File: src/main/java/com/example/demo/service/BookServiceImpl.java [hunk 3]
```diff
@@ -33,4 +33,17 @@ public BookDto save(CreateBookRequestDto bookDto) {
...
         return bookMapper.toDtoList(bookRepository.findAll());
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookDto) {
```
Comment (JJJazl) [inline src/main/java/com/example/demo/service/BookServiceImpl.java:45]: Add `@Transactional` above the class. Spring will create a proxy of your class and each public method will have logic very similar to the one you had implemented in `BookRepositoryImpl`. We need it in order to have 1 transaction during all service method

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
```
Comment (JJJazl) [inline src/main/resources/db/changelog/db.changelog-master.yaml:1]: You can also add a script to insert 2-3 default books to make it easier to test new features in the future

**Review by JJJazl — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/model/Book.java [hunk 5]
```diff
@@ -9,10 +9,14 @@
...
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
Comment (okuzan) [inline src/main/java/com/example/demo/model/Book.java:19]: use SQLRestriction, where is depreated, uppercase TRUE FALSE boolean literals

**Review by okuzan — APPROVED** [review]

**Pull request merged**